src/components/upload/FileUpload.js

```diff
@@ -40,43 +35,65 @@ const FileUpload = () => {
     const [state, setState] = useState("idle");
     const [projectFile, setProjectFile] = useState();
     const [projectName, setProjectName] = useState("");
+    const [open,setOpen] = React.useState(false);
+    const [enableButton] = useState("");
+    const navigate = useNavigate();
 
-    const onSubmit = () => {
-        setState("uploading");
+       const onSubmit = (e) => {
+        setOpen(true);
+        e.preventDefault();
         uploadProjectFile(projectName, projectFile)
-            .then(() => {
-                Notifications.notify("Project created.", "success")
-                setState("idle");
-            })
-            .catch((err) => {
-                Notifications.notify(`Failed to create project.\n${err}`, "error")
-                setState("idle");
-            })
+         if (projectName && projectFile) {
+            Notifications.notify("Project created.", "success")
+            setState("idle");
+            setOpen(true);
+            navigate("/generate", { replace: true });
+        } else if (!projectName && projectFile){
+            Notifications.notify(`Failed to create project.\n`, "error")
+            setState("idle"); 
+            alert('please enter the project name');
+        } else if (projectName && !projectFile){
+          Notifications.notify(`Failed to create project.\n`, "error")
+          setState("idle"); 
+          alert('please upload the project file');
+      }  else {
+           Notifications.notify(`Failed to create project.\n`, "error")
+           setState("idle"); 
+           alert('please upload the project file and enter the project name');
+      }
```

Comment:
```suggestion
        } else if (!projectName && projectFile){
            Notifications.notify(`Please enter the project name`, "error");
            setState("idle"); 
        } else if (projectName && !projectFile){
          Notifications.notify(`Please upload the project file`, "error");
          setState("idle"); 
      }  else {
           Notifications.notify(`Please upload the project file and enter the project name`, "error");
           setState("idle"); 
      }
```
1. Remove trailing newlines in notifications.
2. Use either alerts or notifications. Not both
3. Add semicolon to the end of line